### ISSUE TITLE
[Bug]: PC0031 in combination with if not findfirst then

### ISSUE THREAD

**Issue by StefanMaron** [ALCops/Analyzers#264]
### Analyzer

PlatformCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

PC0031

### Description

The rule fires although the record is only inserted and modified whenever no record can be found. So it's not reading the record and then modifying it with a subset of loaded fields, but it's either or. I'm not sure if that can be easily detected, but I wouldn't say that the load fields here is bad practice.

### AL code to reproduce

```al
local procedure ScheduleJobAction()
    var
        JobQueueEntry: Record "Job Queue Entry";
    begin
        JobQueueEntry.SetLoadFields(ID);
        JobQueueEntry.SetRange("Object Type to Run", JobQueueEntry."Object Type to Run"::Codeunit);
        JobQueueEntry.SetRange("Object ID to Run", Codeunit::SyncRunner);
        if not JobQueueEntry.FindFirst() then begin
            JobQueueEntry.Init();
            JobQueueEntry.Validate(ID, CreateGuid());
            JobQueueEntry.Insert(true);
            JobQueueEntry.Validate("Object Type to Run", JobQueueEntry."Object Type to Run"::Codeunit);
            JobQueueEntry.Validate("Object ID to Run", Codeunit::SyncRunner);
            JobQueueEntry.Validate("Recurring Job", true);
            JobQueueEntry.Validate("Run on Mondays", true);
            JobQueueEntry.Validate("Run on Tuesdays", true);
            JobQueueEntry.Validate("Run on Wednesdays", true);
            JobQueueEntry.Validate("Run on Thursdays", true);
            JobQueueEntry.Validate("Run on Fridays", true);
            JobQueueEntry.Validate("Run on Saturdays", true);
            JobQueueEntry.Validate("Run on Sundays", true);
            JobQueueEntry.Validate("No. of Minutes between Runs", 30);
            JobQueueEntry.Validate(Description, CopyStr(ScheduleJobDescriptionTxt, 1, MaxStrLen(JobQueueEntry.Description)));
            JobQueueEntry.Validate(Status, JobQueueEntry.Status::"On Hold");
            JobQueueEntry.Modify(true);
        end;
        Page.Run(Page::"Job Queue Entry Card", JobQueueEntry);
    end;
```

**Comment by Arthurvdv** [comment]
With the release of version `v0.8.1` of ALCops, this issue should be re solved I believe.

I’m going to close this issue now, but if you have any feedback or run into bugs/false positives, feel free to reopen it (or create a new issue).
